cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -2536,380 +2536,404 @@ mod tests {
-        assert_matches!(parse.to_policy(ast::PolicyID::from_string("id"), &mut errs), Some(_) => {
-            assert!(errs.is_empty(), "{:?}", miette::Report::new(errs));
-        });
+        assert_parse_policy_succeeds(src);
     }
 
     #[test]
     fn show_policy3() {
-        let mut errs = ParseErrors::new();
         let src = r#"
             permit(principal in User::"jane",action,resource);
         "#;
-        let parse = text_to_cst::parse_policy(src).expect("failed parse");
-        println!("{:#}", parse.as_inner().expect("internal parse error"));
-        assert_matches!(parse.to_policy(ast::PolicyID::from_string("id"), &mut errs), Some(_) => {
-            assert!(errs.is_empty(), "{:?}", miette::Report::new(errs));
-        });
+        assert_parse_policy_succeeds(src);
     }
 
     #[test]
     fn show_policy4() {
-        let mut errs = ParseErrors::new();
         let src = r#"
             forbid(principal in User::"jane",action,resource)unless{
                 context.group != "friends"
             };
         "#;
-        let parse = text_to_cst::parse_policy(src).expect("failed parse");
-        println!("{:#}", parse.as_inner().expect("internal parse error"));
-        assert_matches!(parse.to_policy(ast::PolicyID::from_string("id"), &mut errs), Some(_) => {
-            assert!(errs.is_empty(), "{:?}", miette::Report::new(errs));
-        });
+        assert_parse_policy_succeeds(src);
     }
 
     #[test]
     fn policy_annotations() {
         // common use-case
-        let mut errs = ParseErrors::new();
-        let policy = text_to_cst::parse_policy(
+        let policy = assert_parse_policy_succeeds(
             r#"
             @anno("good annotation")permit(principal,action,resource);
         "#,
-        )
-        .expect("should parse")
-        .to_policy(ast::PolicyID::from_string("id"), &mut errs)
-        .unwrap_or_else(|| panic!("failed convert to AST:\n{:?}", miette::Report::new(errs)));
+        );
         assert_matches!(
             policy.annotation(&ast::AnyId::new_unchecked("anno")),
             Some(ast::Annotation { val, .. }) => assert_eq!(val.as_ref(), "good annotation")
         );
 
         // duplication is error
-        let mut errs = ParseErrors::new();
-        let policy = text_to_cst::parse_policy(
+        let errs = assert_parse_policy_fails(
             r#"
             @anno("good annotation")
             @anno2("good annotation")
             @anno("oops, duplicate")
             permit(principal,action,resource);
         "#,
-        )
-        .expect("should parse")
-        .to_policy(ast::PolicyID::from_string("id"), &mut errs);
-        assert_matches!(policy, None => {
-            // annotation duplication (anno)
-            assert_eq!(errs.len(), 1, "{:?}", miette::Report::new(errs));
-        });
+        );
+        // annotation duplication (anno)
+        assert_eq!(errs.len(), 1, "{:?}", miette::Report::new(errs));
```

Comment:
maybe worth improving this check to an `ExpectedErrorMessage`? (Doesn't have to be in this PR)

also for some other cases below